common/djangoapps/contentserver/middleware.py

```diff
@@ -36,14 +39,24 @@ def is_asset_request(self, request):
             request.path.startswith('/' + XASSET_LOCATION_TAG + '/')
             or
             request.path.startswith('/' + AssetLocator.CANONICAL_NAMESPACE)
+            or
+            self.is_versioned_request(request)
         )
 
+    def is_versioned_request(self, request):
+        return request.path.startswith(VERSIONED_ASSETS_PREFIX)
+
     def process_request(self, request):
         """Process the given request"""
         if self.is_asset_request(request):
             # Make sure we can convert this request into a location.
             if AssetLocator.CANONICAL_NAMESPACE in request.path:
                 request.path = request.path.replace('block/', 'block@', 1)
+
+            # If this is a versioned request, chop off the prefix.
+            if self.is_versioned_request(request):
+                request.path = re.sub(VERSIONED_ASSETS_PATTERN, '', request.path)
```

Comment:
@tobz Can we end up in a situation where a person could use this as an attack vector to make it impossible for us to update the image?  Someone could request the image from the CDN using multiple hashes, the CDN would cache the same image for multiple hashes, updates that collide with one of the hashes that has already been cached never make their way to the end user.   More mischief than dangerous, not sure how much we care about it, but it seems like it would be good to make sure the asset at the given hash exists before providing the content.